chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java

```diff
@@ -2032,61 +2034,16 @@ private boolean tryLoadDump(RenderContext context, String fileName, TaskTracker
       }
       return false;
     }
-    try(PushbackInputStream input = new PushbackInputStream(new FileInputStream(dumpFile), 4)) {
-      byte[] magicNumber = new byte[4];
-      input.read(magicNumber, 0, 4);
-      if(Arrays.equals(magicNumber, DUMP_FORMAT_MAGIC_NUMBER)) {
-        // Format with a version number
-        try(DataInputStream dataInput = new DataInputStream(input);
-            TaskTracker.Task task = taskTracker.task("Loading render dump", 2)) {
-          int dumpVersion = dataInput.readInt();
-          if(dumpVersion == 1) {
-            task.update(1);
-            Log.info("Reading render dump " + fileName);
-            int dumpWidth = dataInput.readInt();
-            int dumpHeight = dataInput.readInt();
-            if (dumpWidth != width || dumpHeight != height) {
-              Log.warn("Render dump discarded: incorrect width or height!");
-              return false;
-            }
-            spp = dataInput.readInt();
-            renderTime = dataInput.readLong();
-
-            FloatingPointCompressor.decompress(input, samples);
-          }
-        }
-      } else {
-        // Old format that is a gzipped stream, the header needs to be pushed back
-        input.unread(magicNumber, 0, 4);
-        try (DataInputStream in = new DataInputStream(new GZIPInputStream(input));
-             TaskTracker.Task task = taskTracker.task("Loading render dump", 2)) {
-          task.update(1);
-          Log.info("Reading render dump " + fileName);
-          int dumpWidth = in.readInt();
-          int dumpHeight = in.readInt();
-          if (dumpWidth != width || dumpHeight != height) {
-            Log.warn("Render dump discarded: incorrect width or height!");
-            return false;
-          }
-          spp = in.readInt();
-          renderTime = in.readLong();
 
-          for (int x = 0; x < width; ++x) {
-            task.update(width, x + 1);
-            for (int y = 0; y < height; ++y) {
-              samples[(y * width + x) * 3 + 0] = in.readDouble();
-              samples[(y * width + x) * 3 + 1] = in.readDouble();
-              samples[(y * width + x) * 3 + 2] = in.readDouble();
-              finalizePixel(x, y);
-            }
-          }
-        }
-      }
-    } catch (IOException e) {
+    Log.info("Loading render dump: " + dumpFile);
+    try {
+      RenderDump.load(dumpFile, this, taskTracker);
+    } catch (IOException | IllegalStateException e) {
       // The render dump was possibly corrupt.
-      Log.warn("Failed to load render dump", e);
+      Log.warn("IO exception while loading render dump!", e);
```

Comment:
Don't yell at the user, it's not their fault that there was an error! :smile:

Reply:
I copied the warning message from the other parts of the scene - I guess I did it the wrong way around 😁